### ISSUE TITLE
Use Gridsome with pnpm

### ISSUE THREAD

**Issue by Js-Brecht** [gridsome/gridsome#551]
I just wanted to drop this here in case anybody else is trying to use Gridsome, while using pnpm as their package manager.
* There are some issues with Webpack resolving modules, but that seems to be pretty common when using pnpm, because of the method that webpack uses.
* Gridsome also makes a few assumptions for you about some Vue packages
   * I'm assuming this is because they want users to be locked in to a specific version of the Vue library, to ensure compatibility with the Gridsome framework.
   * As a result, they are not among your project's dependencies in a gridsome scaffolded project.

Btw, this is after scaffolding a new project with `@gridsome/cli`:
```sh
gridsome create <project-name>
```

First thing, we have to tell Webpack how to find its modules.  In your project root folder, you need to include this in `gridsome.server.js`:
```js
const fs = require('fs');
const path = require('path');

let gridsomeModules = fs.realpathSync("node_modules/gridsome");
gridsomeModules = path.resolve(gridsomeModules, "..");

module.exports = function (api) {
  ...
  api.chainWebpack((config) => {
    config.resolveLoader.modules.prepend(gridsomeModules);
    return config;
  });
  ...
}
```

Then there's a few dependencies that need to be loaded.  
* Gridsome uses babel, and babel wants `core-js` somewhere, but it's not available in Gridsome's modules.
   * Needs `"core-js": "^2.6.9"`
      * Note: I had to use version `^2.6.9` because version 3+ doesn't have the es6 modules where babel is expecting them.

Make a pnpmfile.js, and put this hook into it:
```js
// pnpmfile.js
function readPackage(pkg, context) {
  switch(pkg.name) {
    case "gridsome":
      pkg.dependencies = Object.assign({}, pkg.dependencies, {
        "core-js": "^2.6.9",
      });
      break;
  }
  return pkg;
}

module.exports = {
  hooks: {
    readPackage,
  },
};
```

* Because pnpm isn't using a flat node_modules directory like npm/yarn, the libraries you need to develop using Vue are not available to you, so they need to be installed.  Your project needs the following packages:
   * Because of their nature, I put these in `dependencies`:
      * `"vue": "^2.6.10"`
      *  `"vue-meta": "^2.0.5"`
      * `"vue-router": "^3.0.7"`
      * `"vue-template-compiler": "^2.6.10"`

These particular dependencies appear to be needed due to Webpack's quirks.
   * Because they are for building/testing, I put them in `devDependencies`:
      * `"vue-style-loader": "^4.1.2"`
      * `"webpack": "^4.35.3"`
      * `"webpack-hot-middleware": "^2.25.0"`

You can run these two commands:
```
pnpm i vue@^2.6.10 vue-meta@^2.0.5 vue-router@^3.0.7 vue-template-compiler@^2.6.10
pnpm i -D vue-style-loader@^4.1.2 webpack@^4.35.3 webpack-hot-middleware@^2.25.0
```
> An alternative might be to insert them into the main project manifest during install, using the readPackage hook.  Kind of adds another layer of complexity to it, though... easy enough, and more straightforward, to just get these packages using the cli. 

That got me to a successful build, I was able to serve the files, and I was able to use the `gridsome develop` feature.  Of course, this will probably take constant maintenance.  If you ever want to update Gridsome, you're going to have to make sure none of these packages changed, or make sure new ones weren't added that break the build.  Plus, there's going to be a lot of copy-paste going on when you want to spin up a new project.

It would be nice if there were a more solid solution.  Maybe take care of all of this during the project scaffolding, and have an `upgrade` feature that keeps your libraries in sync?  Maybe a yeoman generator?  Even just peer dependencies?

**Comment by hacknug** [comment]
Solving this in core would be so cool ✨

**Comment by hacknug** [comment]
@hjvedvik this wasn't addressed in #1313. That PR fixed things for Yarn v2 but not for pnpm. Just tested creating a new project and I'm still getting errors:

```bash
ERROR  Failed to compile with 4 errors                                                                                                                         13:04:46

These dependencies were not found:

* /Users/nestor/.config/yarn/global/my-website/node_modules/.pnpm/gridsome@0.7.23/node_modules/gridsome/app/entry.client.js in multi webpack/hot/dev-server webpack-hot-middleware/client?name=app&reload=true&noInfo=true ./node_modules/.pnpm/gridsome@0.7.23/node_modules/gridsome/app/entry.client.js ./node_modules/.pnpm/gridsome@0.7.23/node_modules/gridsome/app/entry.sockjs.js
* /Users/nestor/.config/yarn/global/my-website/node_modules/.pnpm/gridsome@0.7.23/node_modules/gridsome/app/entry.sockjs.js in multi webpack/hot/dev-server webpack-hot-middleware/client?name=app&reload=true&noInfo=true ./node_modules/.pnpm/gridsome@0.7.23/node_modules/gridsome/app/entry.client.js ./node_modules/.pnpm/gridsome@0.7.23/node_modules/gridsome/app/entry.sockjs.js
* webpack-hot-middleware/client?name=app&reload=true&noInfo=true in multi webpack/hot/dev-server webpack-hot-middleware/client?name=app&reload=true&noInfo=true ./node_modules/.pnpm/gridsome@0.7.23/node_modules/gridsome/app/entry.client.js ./node_modules/.pnpm/gridsome@0.7.23/node_modules/gridsome/app/entry.sockjs.js
* webpack/hot/dev-server in multi webpack/hot/dev-server webpack-hot-middleware/client?name=app&reload=true&noInfo=true ./node_modules/.pnpm/gridsome@0.7.23/node_modules/gridsome/app/entry.client.js ./node_modules/.pnpm/gridsome@0.7.23/node_modules/gridsome/app/entry.sockjs.js

To install them, you can run: npm install --save /Users/nestor/.config/yarn/global/my-website/node_modules/.pnpm/gridsome@0.7.23/node_modules/gridsome/app/entry.client.js /Users/nestor/.config/yarn/global/my-website/node_modules/.pnpm/gridsome@0.7.23/node_modules/gridsome/app/entry.sockjs.js webpack-hot-middleware/client?name=app&reload=true&noInfo=true webpack/hot/dev-server

```

**Comment by rylanharper** [comment]
Hey everyone, just checking in to see if there any update on this or not really? Anyone tried pnpm with any success using Gridsome?
